engine/GruleEngine.go

```diff
@@ -42,8 +42,9 @@ func NewGruleEngine() *GruleEngine {
 
 // GruleEngine is the engine structure. It has the Execute method to start the engine to work.
 type GruleEngine struct {
-	MaxCycle  uint64
-	Listeners []GruleEngineListener
+	MaxCycle             uint64
+	ErrorOnFailedExecute bool
```

Comment:
Could you use the name like `ReturnErrOnFailedRuleEvaluation`, just to make the variable name become a bit more self explanatory.